packages/core/src/actions.ts

```diff
@@ -157,11 +158,13 @@ export function toActivityDefinition<TContext, TEvent extends EventObject>(
  * @param eventType The event to raise.
  */
 export function raise<TContext, TEvent extends EventObject>(
-  event: Event<TEvent>
-): RaiseAction<TEvent> | SendAction<TContext, AnyEventObject, TEvent> {
-  if (!isString(event)) {
-    return send(event, { to: SpecialTargets.Internal });
+  event: Event<TEvent>,
+  options?: RaiseActionOptions<TContext, TEvent>
+): RaiseAction<TEvent> {
+  if (!isString(event) || options) {
+    return send(event, { ...options, to: SpecialTargets.Internal }) as any;
   }
```

Comment:
IMHO we should stop piggybacking on `send` here. It's especially important right now because you are suppressing the type checker to satisfy the declared return type. I wouldn't be surprised if we miss something that should be handled because of that, now or in the future. At times, we are checking `action.type` after all.